### ISSUE TITLE
WHITELIST_ATTRIBUTES could be Class Method

### ISSUE THREAD

**Issue by viniciuszago** [fidelisrafael/nifty_services#31]
I was thinking that constant `WHITELIST_ATTRIBUTES` could be a Class Method like `record_type`
For example, Instead of:

```ruby
WHITELIST_ATTRIBUTES = [:title, :content]

def record_allowed_attributes
   WHITELIST_ATTRIBUTES
  end
```

Could be:

```ruby

whitelist_attributes :title, :content
```

What do you think?

**Comment by fidelisrafael** [comment]
Good ideia! In the next release I will introduce this, if you want to help fell free to send a PR to nifty_services

**Comment by fidelisrafael** [comment]
Just for reference, I'm thinking in implement this as:

``` ruby
class BaseCrudService
  class << self
    # ServiceClass.whitelist_attributes for outside to access
    attr_reader :whitelist_attributes

    @whitelist_attributes ||= []

    def set_whitelist_attributes(*attrs)
      raise Errors::InvalidWhitelistAttributes if attrs.blank?

      @whitelist_attributes = attrs.map(&:to_sym)

      define_method :record_allowed_attributes do
        @whitelist_attributes
      end

      @whitelist_attributes
    end
  end
end

class PostUpdateService < NiftyServices::BaseUpdateService
  WHITELIST_ATTRIBUTES = [:post, :content].freeze

 set_whitelist_attributes *WHITELIST_ATTRIBUTES
end

# just to make sure
PostUpdateService::WHITELIST_ATTRIBUTES == PostUpdateService.whitelist_attributes
```

**Comment by fidelisrafael** [comment]
@viniciuszago Feel free if you want to contribute with this, just make sure to white tests. (I can help you in this case)

**Comment by fidelisrafael** [comment]
Implemented in #35 \o/
